### PR TITLE
Add vector of vector scatter implementation

### PR DESCRIPTION
**Pull request by permcody**
Slightly more handy interface.

### CODE REVIEW

File: include/parallel/parallel_implementation.h [hunk 1]
```diff
@@ -3191,6 +3191,44 @@ void Communicator::scatter(const std::vector<T> & data,
...
+void Communicator::scatter(const std::vector<std::vector<T> > & data,
+                           std::vector<T> & recv,
+                           const unsigned int root_id,
+                           const bool identical_buffer_sizes) const
+{
+  if (this->size() == 1)
+    {
+      libmesh_assert (!this->rank());
+      libmesh_assert (!root_id);
+      return;
```
Comment (roystgnr) [inline include/parallel/parallel_implementation.h]: This case should be std::copy from data[0] to recv, no?

File: include/parallel/parallel_implementation.h [hunk 2]
```diff
@@ -3191,6 +3191,44 @@ void Communicator::scatter(const std::vector<T> & data,
...
+      for (unsigned int i=0; i < data.size(); ++i)
+        {
+          if (!identical_buffer_sizes)
+            counts[i] = data[i].size();
+          std::copy(data[i].begin(), data[i].end(), std::back_inserter(stacked_data));
+        }
+    }
+
+  if (identical_buffer_sizes)
+    this->scatter(stacked_data, recv, root_id);
```
Comment (roystgnr) [inline include/parallel/parallel_implementation.h]: We should probably have a loop (wrapped in ifndef NDEBUG) which asserts that the user wasn't lying about the identical buffer sizes.

File: include/parallel/parallel_implementation.h (hunk 1 above)
Comment (permcody) [inline include/parallel/parallel_implementation.h]: Yes...  I missed that on the other implementations too. Too much copy and paste.

File: include/parallel/parallel_implementation.h (hunk 2 above)
Comment (permcody) [inline include/parallel/parallel_implementation.h]: Will do

**Comment by roystgnr** [comment]
That's all I saw; good to merge once fixed.

A unit test would be nice, but if we're going to start throwing around blame at people who write code without writing enough test coverage then I'm going to have to flee the country.

**Comment by roystgnr** [comment]
Thanks!  In the future I'd say put the unit test in a separate commit.

**Pull request merged**